File: apps/mobile/android/app/src/main/java/com/houselevi/plus/data/models/ProfileModels.kt
package com.houselevi.plus.data.models

import com.google.gson.annotations.SerializedName

//  Device (GET /auth/devices) 
data class Device(
    @SerializedName("id")          val id: String          = "",
    @SerializedName("deviceId")    val deviceId: String    = "",
    @SerializedName("deviceType")  val deviceType: String  = "",
    @SerializedName("deviceName")  val deviceName: String  = "",
    @SerializedName("os")          val os: String          = "",
    @SerializedName("watchr")     val watchr: String     = "",
    @SerializedName("lastSeenAt")  val lastSeenAt: String  = "",
    @SerializedName("lastSeenIp")  val lastSeenIp: String  = "",
    @SerializedName("loginCount")  val loginCount: Int     = 0,
    @SerializedName("firstSeenAt") val firstSeenAt: String = "",
    @SerializedName("active")      val active: Boolean     = true,
    @SerializedName("trusted")     val trusted: Boolean    = false,
) {
    val displayName: String get() = deviceName.ifBlank { "$deviceType  $os" }
}

data class DevicesResponse(
    @SerializedName("success")    val success: Boolean    = false,
    @SerializedName("devices")    val devices: List<Device> = emptyList(),
    @SerializedName("count")      val count: Int          = 0,
    @SerializedName("maxDevices") val maxDevices: Int     = 3,
)

//  Session (GET /auth/sessions) 
data class UserSession(
    @SerializedName("sessionId")  val sessionId: String   = "",
    @SerializedName("deviceType") val deviceType: String  = "",
    @SerializedName("deviceName") val deviceName: String  = "",
    @SerializedName("lastSeenAt") val lastSeenAt: String  = "",
    @SerializedName("ipAddress")  val ipAddress: String   = "",
    @SerializedName("country")    val country: String     = "",
    @SerializedName("isCurrent")  val isCurrent: Boolean  = false,
)

data class SessionsResponse(
    @SerializedName("success")  val success: Boolean          = false,
    @SerializedName("sessions") val sessions: List<UserSession> = emptyList(),
)

//  Full UserProfile from /auth/me 
data class UserProfile(
    @SerializedName("id")                 val id: String               = "",
    @SerializedName("firstName")          val firstName: String        = "",
    @SerializedName("lastName")           val lastName: String         = "",
    @SerializedName("email")              val email: String            = "",
    @SerializedName("phoneNumber")        val phoneNumber: String      = "",
    @SerializedName("emailVerified")      val emailVerified: Boolean   = false,
    @SerializedName("isPremium")          val isPremium: Boolean       = false,
    @SerializedName("subscriptionStatus") val subscriptionStatus: String = "",
    @SerializedName("isActive")           val isActive: Boolean        = true,
    @SerializedName("role")               val role: String             = "user",
) {
    val fullName: String get() = "$firstName $lastName".trim()
    val initials: String get() = buildString {
        if (firstName.isNotEmpty()) append(firstName.first().uppercaseChar())
        if (lastName.isNotEmpty())  append(lastName.first().uppercaseChar())
    }.ifBlank { email.firstOrNull()?.uppercaseChar()?.toString() ?: "?" }

    val subscriptionLabel: String get() = when (subscriptionStatus.lowercase()) {
        "active"   -> "HL+ Premium"
        "trialing" -> "Free Trial"
        "past_due" -> "Payment Due"
        "canceled" -> "Cancelled"
        else       -> if (isPremium) "HL+ Premium" else "Free"
    }
    val subscriptionActive: Boolean get() =
        subscriptionStatus.lowercase() in listOf("active", "trialing") || isPremium
}

data class ProfileResponse(
    @SerializedName("success") val success: Boolean    = false,
    @SerializedName("user")    val user: UserProfile?  = null,
)

File: apps/mobile/ios/HouseLevi+/Models/ProfileModels.swift
import Foundation

struct Device: Codable, Identifiable, Equatable {
    var id: String = ""
    var deviceId: String = ""; var deviceType: String = ""; var deviceName: String = ""
    var os: String = ""; var browser: String = ""; var lastSeenAt: String = ""
    var lastSeenIp: String = ""; var loginCount: Int = 0; var firstSeenAt: String = ""
    var active: Bool = true; var trusted: Bool = false
    var displayName: String { deviceName.isEmpty ? "\(deviceType)  \(os)" : deviceName }
}

struct UserSession: Codable, Identifiable, Equatable {
    var id: String { sessionId }
    var sessionId: String = ""; var deviceType: String = ""; var deviceName: String = ""
    var lastSeenAt: String = ""; var ipAddress: String = ""; var country: String = ""
    var isCurrent: Bool = false
}

struct UserProfile: Codable, Equatable {
    var id: String = ""; var firstName: String = ""; var lastName: String = ""
    var email: String = ""; var phoneNumber: String = ""; var emailVerified: Bool = false
    var isPremium: Bool = false; var subscriptionStatus: String = ""
    var isActive: Bool = true; var role: String = "user"

    var fullName: String { "\(firstName) \(lastName)".trimmingCharacters(in: .whitespaces) }
    var initials: String {
        var s = ""
        if let f = firstName.first { s += String(f).uppercased() }
        if let l = lastName.first  { s += String(l).uppercased() }
        return s.isEmpty ? (String(email.first ?? "?").uppercased()) : s
    }
    var subscriptionLabel: String {
        switch subscriptionStatus.lowercased() {
        case "active":   return "HL+ Premium"
        case "trialing": return "Free Trial"
        case "past_due": return "Payment Due"
        case "canceled": return "Cancelled"
        default: return isPremium ? "HL+ Premium" : "Free"
        }
    }
    var subscriptionActive: Bool {
        ["active", "trialing"].contains(subscriptionStatus.lowercased()) || isPremium
    }
}

struct DevicesResponse: Codable { var success: Bool = false; var devices: [Device] = []; var count: Int = 0; var maxDevices: Int = 3 }
struct SessionsResponse: Codable { var success: Bool = false; var sessions: [UserSession] = [] }
struct ProfileResponse: Codable { var success: Bool = false; var user: UserProfile? }
